src/integTest/java/com/unosquare/cvgenerator/controller/TemplateControllerIntegrationTest.java

```diff
@@ -0,0 +1,49 @@
+package com.unosquare.cvgenerator.controller;
+
+import com.unosquare.cvgenerator.AbstractWebIntegrationTest;
+import com.unosquare.cvgenerator.dao.TemplateRepository;
+import com.unosquare.cvgenerator.model.entity.TemplateDetail;
+import io.restassured.RestAssured;
+import org.apache.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class TemplateControllerIntegrationTest extends AbstractWebIntegrationTest {
+
+    @Autowired
+    TemplateRepository repository;
```

Comment:
I'd go for either constructor based or property based injection. I think there's a mixture of both in the code base right now.

Reply:
Curious to know if `@RequiredArgsConstructor` would work on a test class